src/geometry.rs

```diff
@@ -816,13 +2899,154 @@ mod tests {
+    }
+
+    #[test]
+    fn test_oriented_edge_is_reverted_returns_true_because_same() {
+        let oriented_edge_one_way = OrientedEdge::new(0, 1);
+        let oriented_edge_other_way = OrientedEdge::new(1, 0);
+        assert!(oriented_edge_one_way.is_reverted(oriented_edge_other_way));
+    }
+
+    #[test]
+    fn test_oriented_edge_is_reverted_returns_false_because_is_same() {
+        let oriented_edge_one_way = OrientedEdge::new(0, 1);
+        let oriented_edge_other_way = OrientedEdge::new(0, 1);
+        assert!(!oriented_edge_one_way.is_reverted(oriented_edge_other_way));
+    }
+
+    #[test]
+    fn test_oriented_edge_is_reverted_returns_false_because_is_different() {
+        let oriented_edge_one_way = OrientedEdge::new(0, 1);
+        let oriented_edge_other_way = OrientedEdge::new(2, 1);
+        assert!(!oriented_edge_one_way.is_reverted(oriented_edge_other_way));
+    }
+
+    #[test]
+    fn test_unoriented_edge_eq_returns_true_because_same() {
+        let unoriented_edge_one_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        let unoriented_edge_other_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        assert_eq!(unoriented_edge_one_way, unoriented_edge_other_way);
+    }
+
+    #[test]
+    fn test_unoriented_edge_eq_returns_true_because_reverted() {
+        let unoriented_edge_one_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        let unoriented_edge_other_way = UnorientedEdge(OrientedEdge::new(1, 0));
+        assert_eq!(unoriented_edge_one_way, unoriented_edge_other_way);
+    }
+
+    #[test]
+    fn test_unoriented_edge_eq_returns_false_because_different() {
+        let unoriented_edge_one_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        let unoriented_edge_other_way = UnorientedEdge(OrientedEdge::new(2, 1));
+        assert_ne!(unoriented_edge_one_way, unoriented_edge_other_way);
+    }
+
+    #[test]
+    fn test_unoriented_edge_hash_returns_true_because_same() {
+        let unoriented_edge_one_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        let unoriented_edge_other_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        let mut hasher_1 = DefaultHasher::new();
+        let mut hasher_2 = DefaultHasher::new();
+        unoriented_edge_one_way.hash(&mut hasher_1);
+        unoriented_edge_other_way.hash(&mut hasher_2);
+        assert_eq!(hasher_1.finish(), hasher_2.finish());
+    }
+
+    #[test]
+    fn test_unoriented_edge_hash_returns_true_because_reverted() {
+        let unoriented_edge_one_way = UnorientedEdge(OrientedEdge::new(0, 1));
+        let unoriented_edge_other_way = UnorientedEdge(OrientedEdge::new(1, 0));
+        let mut hasher_1 = DefaultHasher::new();
+        let mut hasher_2 = DefaultHasher::new();
+        unoriented_edge_one_way.hash(&mut hasher_1);
+        unoriented_edge_other_way.hash(&mut hasher_2);
+        assert_eq!(hasher_1.finish(), hasher_2.finish());
+    }
+
+    #[test]
+    fn test_unoriented_edge_hash_returns_false_because_different() {
```

Comment:
This test is incorrect. The hash can be the same even if data is different. It can start failing randomly when underlying implementation of `DefaultHasher` changes. Let's remove it.

Reply:
How can i test the custom hash function though?
```
impl Hash for UnorientedEdge {
    fn hash<H: Hasher>(&self, state: &mut H) {
        cmp::min(self.0.vertices.0, self.0.vertices.1).hash(state);
        cmp::max(self.0.vertices.0, self.0.vertices.1).hash(state);
    }
}
```

Reply:
With prop tests. https://github.com/BurntSushi/quickcheck

The property we are testing is this: https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq `k1 == k2 -> hash(k1) == hash(k2)`

Let's not do it yet. Enough to add a FIXME for now. But this test needs to be removed.